autowarefoundation/autoware
# Fix/refactoring detection msgs

## Status
DEVELOPMENT

## Description
As part of `autoware_msgs` refactoring, in this PR we refactored only messages concerning **detection**, and created a new messages namespace called `autoware_detection_msgs`.
This was described at https://github.com/CPFL/Autoware/blob/feature/autoware_msgs_refactoring/ros/concerning-autoware-msgs.md, and further discussed at https://github.com/CPFL/Autoware/issues/1428.

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
feature/autoware_msgs_refactoring | [2059](https://github.com/CPFL/Autoware/pull/2059)
fix/refactoring_sync_msgs | [2060](https://github.com/CPFL/Autoware/pull/2060)
fix/refactoring_sensing_msgs | [2061](https://github.com/CPFL/Autoware/pull/2061)
fix/refactoring_localization_msgs | [2062](https://github.com/CPFL/Autoware/pull/2062)
fix/refactoring_planning_msgs | [2063](https://github.com/CPFL/Autoware/pull/2063)
fix/refactoring_socket_msgs | [2064](https://github.com/CPFL/Autoware/pull/2064)

## Todos


## Steps to Test or Reproduce
1. Get a fresh copy of Autoware `develop` branch, ex. `git clone https://github.com/CPFL/Autoware.git -b develop --recurse-submodules [AUTOWARE_PATH]`.
2. Make sure the original builds correctly: run `./catkin_make_release` (follow the steps in [How to build](https://github.com/CPFL/Autoware/wiki/Source-Build) ).
3. Switch to `fix/refactoring_sync_msgs` branch, ex. `git checkout fix/refactoring_sync_msgs` and pull for changes `git pull`.
4. Switch to `fix/refactoring_sensing_msgs` branch, ex. `git checkout fix/refactoring_sensing_msgs` and pull for changes `git pull`.
5. Switch to `fix/refactoring_localization_msgs` branch, ex. `git checkout fix/refactoring_localization_msgs` and pull for changes `git pull`.
6. Switch to `fix/refactoring_planning_msgs` branch, ex. `git checkout fix/refactoring_planning_msgs` and pull for changes `git pull`.
7. Switch to `fix/refactoring_socket_msgs` branch, ex. `git checkout fix/refactoring_socket_msgs` and pull for changes `git pull`.
8. Switch to `fix/refactoring_detection_msgs` branch, ex. `git checkout fix/refactoring_detection_msgs` and pull for changes `git pull`.
9. Make sure refactored version builds correctly: run `./catkin_make_release`.
